python/cuml/dask/linear_model/linear_regression.py

```diff
@@ -14,547 +13,315 @@
+        y_partition_workers = [w for w, xc in y_futures]
+
+        if set(X_partition_workers) != set(y_partition_workers):
+            raise ValueError("""
+              X  and y are not partitioned on the same workers expected \n
+              Linear Regression""")
+
+        self.rnks = dict()
+        rnk_counter = 0
+        worker_to_parts = OrderedDict()
+        for w, p in X_futures:
+            if w not in worker_to_parts:
+                worker_to_parts[w] = []
+            if w not in self.rnks.keys():
+                self.rnks[w] = rnk_counter
+                rnk_counter = rnk_counter + 1
+            worker_to_parts[w].append(p)
+
+        worker_to_parts_y = OrderedDict()
+        for w, p in y_futures:
+            if w not in worker_to_parts_y:
+                worker_to_parts_y[w] = []
+            worker_to_parts_y[w].append(p)
+
+        workers = list(map(lambda x: x[0], X_futures))
+
+        comms = CommsContext(comms_p2p=False)
+        comms.init(workers=workers)
+
+        worker_info = comms.worker_info(comms.worker_addresses)
+
+        key = uuid1()
+        partsToRanks = [(worker_info[wf[0]]["r"], self.client.submit(
+            LinearRegression._func_get_size,
+            wf[1],
+            workers=[wf[0]],
+            key="%s-%s" % (key, idx)).result())
+            for idx, wf in enumerate(X_futures)]
+
+        N = X.shape[1]
+        M = reduce(lambda a, b: a+b, map(lambda x: x[1], partsToRanks))
+
+        key = uuid1()
+        self.linear_models = [(wf[0], self.client.submit(
+            LinearRegression._func_create_model,
+            comms.sessionId,
+            **self.kwargs,
+            workers=[wf[0]],
+            key="%s-%s" % (key, idx)))
+            for idx, wf in enumerate(worker_to_parts.items())]
+
+        key = uuid1()
+        linear_fit = dict([(worker_info[wf[0]]["r"], self.client.submit(
+            LinearRegression._func_fit,
+            wf[1],
+            worker_to_parts[wf[0]],
+            worker_to_parts_y[wf[0]],
+            M, N,
+            partsToRanks,
+            worker_info[wf[0]]["r"],
+            key="%s-%s" % (key, idx),
+            workers=[wf[0]]))
+            for idx, wf in enumerate(self.linear_models)])
+
+        wait(list(linear_fit.values()))
+        raise_exception_from_futures(list(linear_fit.values()))
+
+        comms.destroy()
+
+        self.local_model = self.linear_models[0][1].result()
+        self.coef_ = self.local_model.coef_
+        self.intercept_ = self.local_model.intercept_
+
+    def _fit_with_colocality(self, X, y):
+        input_futures = self.client.sync(extract_colocated_ddf_partitions,
+                                         X, y, self.client)
+        workers = list(input_futures.keys())
+
+        comms = CommsContext(comms_p2p=False)
+        comms.init(workers=workers)
+
+        worker_info = comms.worker_info(comms.worker_addresses)
+
+        N = X.shape[1]
```

Comment:
as in Ridge, would be nice to call these n_rows and n_cols

Reply:
Done.